pkg/services/contexthandler/contexthandler.go

```diff
@@ -138,38 +138,26 @@ func (h *ContextHandler) Middleware(next http.Handler) http.Handler {
 		))
 
 		if h.Cfg.IDResponseHeaderEnabled && reqContext.SignedInUser != nil {
-			namespace, id := getNamespaceAndID(reqContext.SignedInUser)
-			reqContext.Resp.Before(h.addIDHeaderEndOfRequestFunc(namespace, id))
+			reqContext.Resp.Before(h.addIDHeaderEndOfRequestFunc(reqContext.SignedInUser))
 		}
 
 		next.ServeHTTP(w, r)
 	})
 }
 
-// TODO(kalleep): Refactor to user identity.Requester interface and methods after we have backported this
-func getNamespaceAndID(user *user.SignedInUser) (string, string) {
-	var namespace, id string
-	if user.UserID > 0 && user.IsServiceAccount {
-		id = strconv.Itoa(int(user.UserID))
-		namespace = "service-account"
-	} else if user.UserID > 0 {
-		id = strconv.Itoa(int(user.UserID))
-		namespace = "user"
-	} else if user.ApiKeyID > 0 {
-		id = strconv.Itoa(int(user.ApiKeyID))
-		namespace = "api-key"
-	}
-
-	return namespace, id
-}
-
-func (h *ContextHandler) addIDHeaderEndOfRequestFunc(namespace, id string) web.BeforeFunc {
+func (h *ContextHandler) addIDHeaderEndOfRequestFunc(ident identity.Requester) web.BeforeFunc {
 	return func(w web.ResponseWriter) {
 		if w.Written() {
 			return
 		}
 
-		if namespace == "" || id == "" {
+		namespace, id := ident.GetNamespacedID()
+		if !identity.IsNamespace(
+			namespace,
+			identity.NamespaceUser,
+			identity.NamespaceServiceAccount,
+			identity.NamespaceAPIKey,
+		) || id == "0" {
```

Comment:
nit: we previously weren't checking for `id == "0"`, but I guess it's fine to add.

Reply:
No but we checked if id was bigger then 0. Unauthenticated users have `user:0` namespace so this is just to remove those